src/assets/scripts/client/aircraft/FlightManagementSystem/RouteModel.js

```diff
@@ -812,31 +808,55 @@ export default class RouteModel extends BaseModel {
 
     // ------------------------------ PRIVATE ------------------------------
 
-    _appendRouteModelBeginningAtWaypointName(waypointName, routeModel) {
-        const indexOfDivergentLeg = this._findIndexOfLegContainingWaypointName(waypointName);
+    /**
+     * Append a provided route model into this RouteModel
```

Comment:
I think you mean:
> Append this `RouteModel` with a provided `RouteModel`

Reply:
We should also add some notes in here about how this is going to mutate the existing `legCollection`

Reply:
Changed to

```
/**
 * Append a provided route model onto the end of this RouteModel
 *
 * This method only serves to call the method that contains the appropriate logic
 * based on the type of leg in which the divergent waypoint resides, since this
 * heavily weighs in to how the merging of the routes should be done. Note that
 * the #_legCollection will be mutated in this process.
```

done.